http/client.go

```diff
@@ -0,0 +1,39 @@
+package http
+
+import (
+	"crypto/tls"
+	"errors"
+	"net/http"
+)
+
+var ErrNilCert = errors.New("nil cert")
+
+// ClientWithCert injects cert for mTLS into a copy of client.
+// Transports and TLS configs are created (if nil) or cloned as needed.
+// If client is nil the default HTTP client will be used.
+func ClientWithCert(client *http.Client, cert *tls.Certificate) (*http.Client, error) {
+	if cert == nil {
+		return client, ErrNilCert
+	}
+
+	if client == nil {
+		client = http.DefaultClient
+	}
+	clientCopy := *client
+	client = &clientCopy
```

Comment:
Is this just so you can hand back a pointer to the client? Funky pattern (to me 🤣 )

Reply:
Yeah it's ln22: "assign a _copy_ of the object at that pointer" then ln23: "assign the pointer to my newly copied object to my object"

Basically a way to modify a copy of the client without disturbing the original client. Pointers.. were a mistake. ;P